### PR TITLE
#37116 Change return data in getMenu function

### PR DESCRIPTION
**Pull request by dawiddczaja**

### CODE REVIEW

File: Block/Menu.php [hunk 1]
```diff
@@ -84,20 +84,33 @@ protected function getCacheLifetime()
     /**
      * @return \Snowdog\Menu\Model\Menu
      */
-    public function getMenu()
+    protected function loadMenu()
```
Comment (DariuszMaciejewski) [inline Block/Menu.php]: Is there any reason why it is `protected` instead of `private`?

File: view/frontend/templates/menu.phtml [hunk 2]
```diff
@@ -33,3 +34,4 @@
     </ul>
     <div class="<?= $menuClass ?>__mobile-bg"></div>
 </nav>
+<?php endif; ?>
```
Comment (DariuszMaciejewski) [inline view/frontend/templates/menu.phtml]: Add a newline

File: view/frontend/templates/menu/sub_menu.phtml [hunk 3]
```diff
@@ -54,3 +55,4 @@ $wrapperAttributes = [
         </li>
     <?php endforeach ?>
 </ul>
+<?php endif; ?>
```
Comment (DariuszMaciejewski) [inline view/frontend/templates/menu/sub_menu.phtml]: Add a newline

File: Block/Menu.php (hunk 1 above)
Comment (dawiddczaja) [reply Block/Menu.php]: No, I changed to private

**Comment by DariuszMaciejewski** [comment]
squash and merge

**Comment by Koc** [comment]
>   Block/Menu.php 100644 → 100755 

@dawiddczaja please do not change files permissions.

File: view/frontend/templates/menu.phtml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 <?php /** @var \Snowdog\Menu\Block\Menu $block */ ?>
+<?php if ($block->getMenu()): ?>
 <?php $menuClass = $block->getMenu()->getCssClass() ?>
```
Comment (Koc) [inline view/frontend/templates/menu.phtml]: Please ident with 4 spaces if body

Comment (dawiddczaja) [reply view/frontend/templates/menu.phtml]: Added

Comment (dawiddczaja) [reply view/frontend/templates/menu.phtml]: Added

**Pull request closed without merging**